src/crud/movies.py

```
@@ -0,0 +1,172 @@
+            languages_obj.append(existing_languages[language_name])
+        else:
+            new_language = LanguageModel(name=language_name)
+            db.add(new_language)
+            await db.flush()
+            languages_obj.append(new_language)
+
+    new_movie.languages = languages_obj
+
+    country_code = movie.country.upper()
+
+    result = await db.execute(
+        select(CountryModel).where(CountryModel.code == country_code)
+    )
+    existing_country = result.scalar_one_or_none()
+
+    if existing_country:
+        new_movie.country = existing_country
+    else:
+        new_country = CountryModel(
+            code=country_code,
+            name=None
+        )
+        db.add(new_country)
+        await db.flush()
+        new_movie.country = new_country
+
+    db.add(new_movie)
+    await db.commit()
+    stmt = select(MovieModel).options(
+        selectinload(MovieModel.country),
+        selectinload(MovieModel.genres),
+        selectinload(MovieModel.actors),
+        selectinload(MovieModel.languages),
+    ).where(MovieModel.id == new_movie.id)
+
+    result = await db.execute(stmt)
+    movie_with_relations = result.scalar_one()
+
+    return movie_with_relations
+
+
+async def get_movie(db: AsyncSession, movie_id: int):
+    result = await db.execute(select(MovieModel).options(
+        selectinload(MovieModel.country),
+        selectinload(MovieModel.genres),
+        selectinload(MovieModel.actors),
+        selectinload(MovieModel.languages)).where(MovieModel.id == movie_id))
+    movie = result.scalar_one_or_none()
+    return movie
+
+
+async def delete_movie(db: AsyncSession, movie_id: int):
+    result = await db.execute(select(MovieModel).options(
+        selectinload(MovieModel.country),
+        selectinload(MovieModel.genres),
+        selectinload(MovieModel.actors),
+        selectinload(MovieModel.languages)).where(MovieModel.id == movie_id))
+    movie = result.scalar_one_or_none()
+
+    await db.delete(movie)
```

Issue: There is no check to see if `movie` is `None` before calling `await db.delete(movie)`. If the movie does not exist, this will raise an error. Please add a check and raise an HTTPException with status 404 if the movie is not found.